cdk/ecs/lib/app.ts

```diff
@@ -0,0 +1,122 @@
+        this.createStacks();
+        this.createServers();
+        this.createApiGateway();
+        this.runServices();
+        this.generateTraffic();
+        this.app.synth();
+    }
+
+    private createStacks(): void {
+        const networkStack = new NetworkStack(this.app, 'NetworkStack');
+
+        this.logStack = new LogStack(this.app, 'LogStack');
+
+        this.loadbalancerStack = new LoadBalancerStack(this.app, 'LoadBalancerStack', {
+            vpc: networkStack.vpc,
+            securityGroup: networkStack.albSecurityGroup,
+        });
+
+        this.rdsDatabaseStack = new RdsDatabaseStack(this.app, 'RdsDatabaseStack', {
+            vpc: networkStack.vpc,
+            rdsSecurityGroup: networkStack.rdsSecurityGroup,
+        });
+
+        const iamRolesStack = new IamRolesStack(this.app, 'IamRolesStack');
+
+        // Grant ecsTaskRole access to database
+        this.rdsDatabaseStack.dbSecret.grantRead(iamRolesStack.ecsTaskRole);
+        this.rdsDatabaseStack.dbSecret.grantWrite(iamRolesStack.ecsTaskRole);
+
+        this.serviceDiscoveryStack = new ServiceDiscoveryStack(this.app, 'ServiceDiscoveryStack', {
+            vpc: networkStack.vpc,
+        });
+
+        this.ecsClusterStack = new EcsClusterStack(this.app, 'EcsClusterStack', {
+            vpc: networkStack.vpc,
+            securityGroups: [networkStack.ecsSecurityGroup],
+            ecsTaskRole: iamRolesStack.ecsTaskRole,
+            ecsTaskExecutionRole: iamRolesStack.ecsTaskExecutionRole,
+            subnets: networkStack.vpc.publicSubnets,
+            serviceDiscoveryStack: this.serviceDiscoveryStack,
+            logStack: this.logStack,
+        });
+    }
+
+    private createServers() {
+        this.ecsClusterStack.createConfigServer();
+        this.ecsClusterStack.createDiscoveryServer();
+        this.ecsClusterStack.createAdminServer();
+    }
+
+    private createApiGateway() {
+        this.ecsClusterStack.createApiGateway(
+            this.loadbalancerStack.loadBalancer.loadBalancerDnsName,
+            this.adotJavaImageTag,
+            this.loadbalancerStack.targetGroup,
+        );
+    }
+
+    private runServices() {
+        this.ecsClusterStack.runVetsService(this.adotJavaImageTag);
+        this.ecsClusterStack.runCustomersService(this.adotJavaImageTag);
+        this.ecsClusterStack.runVisitsService(this.adotJavaImageTag);
+        this.ecsClusterStack.runInsuranceService(
+            this.adotPythonImageTag,
+            this.rdsDatabaseStack.dbSecret,
+            this.rdsDatabaseStack.rdsInstance.dbInstanceEndpointAddress,
+        );
+        this.ecsClusterStack.runBillingService(
+            this.adotPythonImageTag,
+            this.rdsDatabaseStack.dbSecret,
+            this.rdsDatabaseStack.rdsInstance.dbInstanceEndpointAddress,
+        );
+    }
+
+    private generateTraffic() {
+        this.ecsClusterStack.generateTraffic(this.loadbalancerStack.loadBalancer.loadBalancerDnsName);
```

Comment:
I feel it's a little strange to use the stack this way. Usually a stack is used to create resources and it's usually self contained. This means that it usually create relevant resources by itself. Here the stack is providing an API and you are asking a caller to invoke the API to create the traffic generator. This is not I normally see in CDK code.